k4FWCore/include/k4FWCore/DataWrapper.h

```diff
@@ -69,8 +69,9 @@ template <class T> class GAUDI_API DataWrapper : public DataWrapperBase {
 };
 
 template <class T> DataWrapper<T>::~DataWrapper() {
-  if (is_owner && !m_data)
+  if (is_owner && m_data) {
     delete m_data;
+  }
```

Comment:
Is it necessary to check the pointer? Deleting a null is defined in the standard
```suggestion
  if (is_owner) {
    delete m_data;
  }
```